diff --git a/dist/emmet.cjs b/dist/emmet.cjs
index 8d295bced45dd325629717900c308ef0ff6811cf..7e92da2ab7d0c7ec5726a7de419ada4e27d5d359 100644
--- a/dist/emmet.cjs
+++ b/dist/emmet.cjs
@@ -1242,7 +1242,7 @@ function convertAttribute(node, state) {
     let valueType = node.expression ? 'expression' : 'raw';
     let value;
     const name = node.name && stringifyName(node.name, state);
-    if (name && name[0] === '!') {
+    if (name && name[0] === '-') {
         implied = true;
     }
     if (name && name[name.length - 1] === '.') {
diff --git a/dist/emmet.es.js b/dist/emmet.es.js
index 1fa26984ab9fc6030ed748b3890ed4681d04e5c4..724ba257032b6b623eea08eeac1bb89466bc291a 100644
--- a/dist/emmet.es.js
+++ b/dist/emmet.es.js
@@ -1238,7 +1238,7 @@ function convertAttribute(node, state) {
     let valueType = node.expression ? 'expression' : 'raw';
     let value;
     const name = node.name && stringifyName(node.name, state);
-    if (name && name[0] === '!') {
+    if (name && name[0] === '-') {
         implied = true;
     }
     if (name && name[name.length - 1] === '.') {
